core/drivers/firewall/stm32_rif.c

```diff
@@ -118,22 +110,15 @@ TEE_Result stm32_rif_acquire_semaphore(vaddr_t addr,
 TEE_Result stm32_rif_release_semaphore(vaddr_t addr,
 				       unsigned int nb_cid_supp)
 {
-	unsigned int msb_nb_cid_supp = sizeof(nb_cid_supp) * 8 -
-				       __builtin_clz(nb_cid_supp | 1) - 1;
-	uint32_t scid_mask = 0;
+	uint32_t scid_mask = get_scid_mask(nb_cid_supp);
 
-	/* SCID bitfield highend can't be > SCID_SHIFT + MAX_CID_BITFIELD */
-	assert(msb_nb_cid_supp <= MAX_CID_BITFIELD);
-
-	if (stm32_rif_is_semaphore_available(addr))
+	if (stm32_rif_semaphore_is_available(addr))
```

Comment:
A CPU might not be alone in a compartment.
While I agree stm32_rif_release_semaphore() should be released after stm32_rif_acquire_semaphore() and no other software component in the same CID should access the same resource, it can happen.
If you agree, I'd prefer to keep it this way